homeassistant/components/google_pubsub/__init__.py

```diff
@@ -45,9 +45,12 @@ def setup(hass: HomeAssistant, yaml_config: dict[str, Any]):
     config = yaml_config[DOMAIN]
     project_id = config[CONF_PROJECT_ID]
     topic_name = config[CONF_TOPIC_NAME]
-    service_principal_path = os.path.join(
-        hass.config.config_dir, config[CONF_SERVICE_PRINCIPAL]
-    )
+    if hass.config.config_dir:
```

Comment:
We should use this instead:
https://github.com/home-assistant/core/blob/6d7ad8903f74959c6e3309907c77bc256bff5555/homeassistant/core.py#L1589

Then we don't need this check. It's not correct either.

Reply:
#54744